### PR TITLE
Bugfix/pettingzoo agent id concatenation

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Agent IDs were not being concatenated to observations for PettingZoo environments since the environment with agent IDs added was called `env` but the `make_environment` function was returning `environment`
## Why?
To fix the bug. 
## How?
Changed the variable name. 
## Extra
Raise a `NotImplementedError` in the `ConcatAgentIdToObservation` when trying to concatenate agent IDs to a non-vector observation. 
Has the same error now in the `ConcatAgentIdToObservation` as mentioned [here](https://github.com/instadeepai/Mava/pull/804#discussion_r1006432340) since certain PettingZoo environments do not return extras. So this depends on that PR.

### CODE REVIEW

**Review by DriesSmit — APPROVED** [review]
Great catch! :fire: Tanks @RuanJohn :slightly_smiling_face:

File: mava/utils/environments/pettingzoo_utils.py [hunk 1]
```diff
@@ -131,6 +131,6 @@ def make_environment(
         raise Exception("Pettingzoo is not installed.")
 
     if concat_agent_id:
-        env = ConcatAgentIdToObservation(env)
+        environment = ConcatAgentIdToObservation(environment)  # type: ignore
```
Comment (ulricharmel) [inline mava/utils/environments/pettingzoo_utils.py:134]: Thanks @RuanJohn . Okay cool I didn't know about the # type ignore flag. I actually added a fix to this in the frame staking PR but I had to define env as type Any to bypass the pre-commit type error.

**Review by ulricharmel — APPROVED** [review]
Thanks @RuanJohn

File: mava/utils/environments/pettingzoo_utils.py (hunk 1 above)
Comment (RuanJohn) [reply mava/utils/environments/pettingzoo_utils.py:134]: Thank you @ulricharmel, I try to use it very sparingly :sweat_smile: but in this case all my ideas for getting around the type checking made the code much less clean to read. The Any idea is also good.

**Pull request merged**